compare_template_text/normalize_template_text.py

```diff
@@ -32,67 +33,42 @@ def remove_omitable_text(self):
         which is optional and present matches and gets replaced by '' while the text that doesn't match will 
         remain as it is thus giving an unmatch. """
          
-        lists = re.findall('(?<=\<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
+        lists = re.findall('(?<=<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
```

Comment:
Since you have escaped the template string, this search may need to be updated.  I would suggest setting a breakpoint and examining the self.template_string or printing the self.template_string to see if the pattern has changed.